pkg/chains/annotations/annotations.go

```diff
@@ -127,13 +139,7 @@ func AddAnnotation(ctx context.Context, obj objects.TektonObject, ps versioned.I
 		mergedAnnotations[k] = v
 	}
 
-	// Add the specific key-value pair, again it must be chains annotation
-	if !strings.HasPrefix(key, ChainsAnnotationPrefix) {
-		return fmt.Errorf("invalid annotation key %q: all annotations must have prefix %q", key, ChainsAnnotationPrefix)
```

Comment:
I think we still keep this validation just make sure that we continue to be defensive wrt there being a prefix.

Reply:
The `key` variable does not exist anymore, that's why the check was removed.
Initially the same function could have received annotations in two ways, a key/value pair or a list of annotations (a map). To simplify this, I removed the first options and instead added a helper function `mergeAnnotations` which deals with that scenario. And the above check still exists as part of the validation of the annotations provided as a map.